packages/connect-wallet-modal/src/components/Ledger/LedgerAccountScreen.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import styled from '@reef-knot/ui-react/styled-wrapper';
+import { Pagination, Stack, StackItem } from '@lidofinance/lido-ui';
+import { AccountButton, AccountButtonSkeleton } from './LedgerAccountButton';
+import { useLedgerAccounts, useLedgerContext } from './hooks';
+import { getFirstIndexOnPage, saveLedgerDerivationPath } from './helpers';
+import { Metrics } from '../WalletsModal';
+import { useConnectorLedger } from '@reef-knot/web3-react';
+import { LedgerDerivationPathSelect } from './LedgerDerivationPathSelect';
+import { AccountsStorage } from './types';
+import { DERIVATION_PATHS } from './constants';
+
+const BoxWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  margin-top: 24px;
+  padding-bottom: 20px;
+`;
+
+type Props = {
+  metrics?: Metrics;
+  closeScreen?: () => void;
+};
+
+const ACCOUNTS_PER_PAGE = 5;
+const ACCOUNTS_TOTAL_PAGES = 4;
+
+export const LedgerAccountScreen: FC<Props> = ({ metrics, closeScreen }) => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [derivationPathTemplate, setDerivationPathTemplate] = useState<string>(
+    DERIVATION_PATHS[0].template,
+  );
+  const [accountsStorage, setAccountsStorage] = useState<AccountsStorage>();
+
+  const { setError, disconnectTransport } = useLedgerContext();
+
+  const accountsForPage = useLedgerAccounts({
+    accountsStorage,
+    derivationPathTemplate,
+    accountsPerPage: ACCOUNTS_PER_PAGE,
+    currentPage,
+  });
+
+  useEffect(() => {
+    if (Object.values(accountsForPage).length === ACCOUNTS_PER_PAGE) {
+      const { pathTemplate } = Object.values(accountsForPage)[0];
+
+      setAccountsStorage({
+        ...accountsStorage,
+        [pathTemplate]: {
+          ...accountsStorage?.[pathTemplate],
+          ...accountsForPage,
+        },
+      });
+    }
+  }, [accountsForPage, accountsStorage, derivationPathTemplate]);
+
+  const onConnectLedger = metrics?.events?.connect?.handlers.onConnectLedger;
+  const { connect, connector } = useConnectorLedger({
+    onConnect: onConnectLedger,
+  });
+
+  useEffect(() => {
+    if (!connect || !connector?.isSupported()) {
+      setError(
+        new Error(
+          "Your browser doesn't support direct connection with Ledger. Please, try another browser.",
+        ),
+      );
+    }
+  }, [connect, connector, setError]);
```

Comment:
setters are not needed in dependencies, actually
```suggestion
  }, [connect, connector]);
```

Reply:
Then eslint will always warn about it and highlight this place.

Reply:
Strange... it should not, but OK. maybe old version of eslint plugin is used?